### ISSUE TITLE
Detection of generated code

### ISSUE THREAD

**Issue by SRechenberger** [fwcd/KotlinLanguageServer#112]
Given the following example
```kotlin
import kotlinx.serialization.Serializable

@Serializable
data class X(val a: Int)

val foo = X.serializer()
```
VS Code can not resolve the reference `serializer`, although the compiler can build it without problems.
I guess the problem is, that the method is generated at compile time, and does not exists, as the language server needs to know of it.

**Comment by belenot** [comment]
+1
Maybe there is option to set to target generated classes for lsp?

**Comment by carlospaulino** [comment]
@SRechenberger @belenot  Did you ever find a workaround to get the language server to work with generated sources?

cc: @rckgomz

**Comment by danielfigueiredo** [comment]
Yeah I can get it to work with everything, but the generated classes. Some tools like Netflix Dsg (for graphql) among others generate the java classes under `target` and they are invisible to the LSP
There is an option to specify jars, but not classes to include. Ideally it should detect `target` folder automatically? I'm not well versed in java/kotlin, but seems to be a convention
